### ISSUE TITLE
Upgrade from Free to Premium Failed!

### ISSUE THREAD

**Issue by HelloIsaac** [kangarko/ChatControl#31]
As promised, just purchased Premium and attempted an upgrade. It went mental, so I had to revert to my free 5.2.1 and previous configuration:

http://pastebin.com/BtndwyXq

I tried various steps to resolve the problem, and I get the sense it might be caused by jumping from 5.2.1 to 6.4.something, but you'll know better than I do! I tried deleting my original settings file and allowing it to load its own, but this just meant that no chat came through at all (even when formatting was on).

Hopefully there's something that can be done, because I'm looking forward to upgrading!

Thanks a lot :)

**Comment by kangarko** [comment]
Sorry I am in a hurry, I will investigate ASAP. Please stay on the free version for the time being.

**Comment by kangarko** [comment]
Could you send me the Messages.Timed part from your settings.yml please? The error is coming from there.

**Comment by HelloIsaac** [comment]
Oh, interesting. I've got a fairly long set of timed messages, so here's a pastebin:

http://pastebin.com/R7HLiMte

Thanks so much for looking into it, it's really appreciated!

**Comment by kangarko** [comment]
Hey,

according to the https://yaml-online-parser.appspot.com
the sentences containing a colon, ':', are parsed as a list. The parser only expect a simple message and therefore the error is thrown.

Try this version (I've put the problematic sentences in quotes: "").
http://pastebin.com/isH851Ar

**Comment by HelloIsaac** [comment]
Thanks for going through and doing that!

I first tried replacing my original list with this one, but that threw a few YML errors. I changed the spacing, and this issue resolved itself, but another 'Messages' section generated above the one I'd customised, pre-filled with the default stuff. 

I deleted mine, and incorporated the messages from it into the one that had just generated (this is all still on the 5.2.1 version). It was working fine at this point, displaying server messages properly with the slightly different formatting in the .yml file.

I shut the server and dragged in the new .jar - updating gave me this in the console http://pastebin.com/YQ76ewkY which is what I anticipated.

However, the server crashed immediately after this and went into a boot loop that woudn't resolve until I downgraded back to the free version and restored my original Chat Control files (pre-upgrade).

It got exactly this far in the boot process and then shut down, trying to start-up again: http://pastebin.com/PHdFekfi

Some sort of Factions problem?

Thanks, again, for looking into this.

**Comment by kangarko** [comment]
Hey, I apologies for the inconveniences. According to the first pastebin log, the upgrade went fine and the file has been saved properly. I don't see how the crash could happen, I never experienced such thing. 

Can you please explain the second log? What I see is only ChatControl's messages saying the plugin is being loaded. What messages followed next? I would more believe that the crash was caused because of another plugin rather than ChatControl. 

To exclude that the Factions cause the problem, try removing it temporary from your server and boot it up with the premium CHC version. If it won't help, leave the Factions disabled and try booting server with the default ChatControl configuration (just rename chatcontrol folder in plugins folder to something else). Then, please post me the whole log (if you do not wish to publish private data you can pm me at spigot)

Please note that the free version has not the configuration parser of the quality of the premium version. It lacks more advanced integrity checks and therefore sometimes tends to oversee a malformed configuration.

**Comment by HelloIsaac** [comment]
No worries, Kangarko. Sorry for the delay in getting back to you.

The second log is the very last bit before the server shut down. Factions hasn't been updated in a while (but is on the newest version), and so only ChatControl had changed here which makes me wonder if it's a Factions/ChatControl incompatibility that was causing the server to enter a boot-loop?

I'll try disabling Factions and seeing if that helps, and get back to you :)

**Comment by HelloIsaac** [comment]
So, I've tried a few things, and still not quite successful:

1. First, I updated Factions to the latest version. This broke compatibility with the free version of ChatControl that I'm using (5.2.1), and so I can no longer display Faction Prefixes in chat with the following configuration which had been working previously:

Chat:
  Formatter:
    Enabled: true
    Message_Format: '{factions_relcolor}{factions_name} %pl_prefix%displayname:%pl_suffix %message'

(N.B. When I briefly made a ticket here about how to add Faction tags to chat on the free version, you suggested %faction(s?), I think, which didn't work! The above solution did work, but no longer does with the update to Factions. (I presume that's what has changed, anyway).

It now displays the chat quite literally as, for example:

"{factions_relcolor}{factions_name} [God] Zeus: blah blah blah"

I've just removed Faction prefixes from chat for now, but it'd be good to get that working again somehow! Presumably when this update works properly.


2. I tried disabling Factions, renaming the (free) ChatControl folder, installing ChatControl (6.4.3), and letting it generate its own config files anew.

This worked almost perfectly. I could use the new features like /chc announcement for the first time, and booting again with Factions enabled did not affect any aspect of the plugin's functionality.

Except that, in both cases, the formatter simply wasn't doing anything. I was getting the default '<Name>: Message' regardless of what I put into the configuration file (and I'd made sure to change 'enabled' to 'true' for the formatter).

I get the sense, then, that the only problem now is that something is stopping the formatter from working. I imagine the Factions prefix issue would be fixed by being on the latest premium version and just using the %factions tag, or whatever it is.

Here are some logs:

- v5.2.1 (free CC) with latest Factions: http://pastebin.com/8FH06En2

- v6.4.3 (premium) loading for the first time, and with no issues: http://pastebin.com/jC5B0u7s
(Note: Formatter did not work after this perfect load)

- Error message related to Factions the next time I booted 6.4.3: http://pastebin.com/mMknzWie

---------------

Sorry this is so long, but I wanted to give you as much information as possible!

The TL;DR is this: 

- Update seems to work fine, but formatter is impossible to enable. 

- Latest Factions update seems to have broken Faction compatibility with the free version. 

- Some weird error to do with fetching the Faction prefix when I was trying to get it to work. Note that this isn't what's causing the initial problem, because even without Factions stuff in the formatter, it still isn't changing anything (I tried things like "Bob %message" and still didn't get anything)

Thanks again!

**Comment by HelloIsaac** [comment]
Deleted my previous message, because, with a little more tinkering, I have a much more concise update:

I deleted Factions, upgraded to a clean installation (removing the old ChatControl stuff), and it worked absolutely fine.

I then added Factions, rebooted, and there was incompatibility. Chat was just displaying Faction tags and nothing else.

Finally, I re-enabled Factions, and, for no obvious reason, they're now working together! This is the format I'm using:

        Message_Format: '{factions_relcolor}[{factions_name}] %pl_prefix%displayname:%pl_suffix %message'

Which seems to work and has the benefit of colouring the factions which the %faction tag does not.

All in all, this is fixed, but I have no idea why. I just added the auto-messages using "s before and after, and that's also working perfectly. Hopefully, this was just as weird issue that only I come across, but, if other people do also, let me know and I can give you more information.

Thanks so much for all your help, I'm really glad to have supported such an excellent developer!

**Comment by kangarko** [comment]
Let me thank you for your feedback, I really appreciate it. Just curious, might I ask from where the '{factions_relcolor}' variable is coming from, as it is not inbuilt in ChatControl itself?

**Comment by HelloIsaac** [comment]
That's native to Factions, I think. 

I'm not sure how, because I don't know nearly enough about this stuff to guess, but I think my priority combination (chatcontrol to HIGHEST; Factions chat formatting to NORMAL) seems to have made it so that ChatControl allows me to plug in the Factions values ('{factions_relcolor}[{factions_name}]') and can still parse them properly.

The relcolor variable generates a &colourcode depending on one's own relation to the given Faction, so it's just a useful thing to have added, and seems to be working!

**Comment by kangarko** [comment]
Okay thanks for the feedback, I didn't know think that such thing is possible to achieve.
